Makefile

```diff
@@ -22,26 +28,20 @@ all: common otelcontribcol
 e2e-test: otelcontribcol
 	$(MAKE) -C testbed runtests
 
-.PHONY: precommit
-precommit:
-	$(MAKE) gotidy
-	$(MAKE) ci
-
 .PHONY: test-with-cover
-test-with-cover:
+unit-tests-with-cover:
 	@echo Verifying that all packages have test files to count in coverage
 	@scripts/check-test-files.sh $(subst github.com/open-telemetry/opentelemetry-collector-contrib/,./,$(ALL_PKGS))
-	@echo pre-compiling tests
-	set -e; for dir in $(ALL_MODULES); do \
-	  echo "go test ./... + coverage in $${dir}"; \
-	  (cd "$${dir}" && \
-	    $(GOTEST) $(GOTEST_OPT_WITH_COVERAGE) ./... && \
-	 	go tool cover -html=coverage.txt -o coverage.html ); \
-	done
+	@$(MAKE) for-all CMD="make do-unit-tests-with-cover"
```

Comment:
- Why do we want to run each module separately now?
- Does this mean we'll need to upload coverage reports from each module to coveralls now?

Reply:
> * Why do we want to run each module separately now?
It was already being done separately, it's just that the looping was done in bash inside the makefile vs. calling for-all. Thought it'd be better to use for-all and also have the make target available in common so you can run it on an individual module if desired.

> * Does this mean we'll need to upload coverage reports from each module to coveralls now?

I think that was already the case. It already generated a coverage.txt inside each module which codecov collects at the end of circleci run.

Reply:
I see. For some reason I thought we used to run a single `go test ./...` command to test the entire repo in a single run.